src/Cardano/Wallet/Kernel/Wallet.hs

```diff
@@ -0,0 +1,123 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE ExistentialQuantification #-}
+{-# LANGUAGE TypeOperators #-}
+
+-- |
+-- Copyright: © 2018-2019 IOHK
+-- License: MIT
+--
+-- | Module provides internal implementation of
+-- | wallet api
+
+module Cardano.Wallet.Kernel.Wallet where
+
+import Prelude
+
+import Cardano.Wallet
+    ( Wallet )
+import Cardano.Wallet.AddressDiscovery
+    ( AddressPoolGap )
+import Cardano.Wallet.Mnemonic
+    ( Mnemonic )
+import Cardano.Wallet.Primitive
+    ( Coin )
+import Crypto.Encoding.BIP39
+    ( ValidMnemonicSentence )
+import Data.ByteArray
+    ( ScrubbedBytes )
+import Data.Text
+    ( Text )
+import Data.Time.Units
+    ( Microsecond )
+import Fmt
+    ( Buildable (..), fmt )
+
+
+-- | Errors
+data CreateWalletError = CreateWalletError
+
+instance Buildable CreateWalletError where
+    build CreateWalletError =
+        mempty
+        <> "CreateWalletError"
+
+instance Show CreateWalletError where
+    show = fmt . build
+
+data GetWalletError = GetWalletError
+
+instance Buildable GetWalletError where
+    build GetWalletError =
+        mempty
+        <> "GetWalletError"
+
+instance Show GetWalletError where
+    show = fmt . build
+
+
+-- | Types
+newtype CreateWallet = CreateWallet NewWallet
+
+data NewWallet =
+    forall mw . ValidMnemonicSentence mw => NewWallet
+    {
+      _mnemonicSentence
```

Comment:
Did we settle the decision on how to generate lenses? TH?

Reply:
good question

Reply:
`Generic` through https://hackage.haskell.org/package/generic-lens :+1:

Reply:
Though, lenses is a topic we have to discuss in our coding standards before we start using them.

Reply:
If we do...

Reply:
added `Generic` derivation to handle `generic-lens`